library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -115,6 +134,11 @@ private void addHeader(DownloadInfo info, String header, String value) {
             info.mRequestHeaders.add(Pair.create(header, value));
         }
 
+        private void setDownloadUris(DownloadInfo info) {
+            info.allDownloadsUri = ContentUris.withAppendedId(Downloads.Impl.ALL_DOWNLOADS_CONTENT_URI, info.mId);
```

Comment:
:x: this change breaks the current behaviour.

Let's try and do minimal changes to the DownloadManager or be extra careful when doing them.

This breaks the behaviour as now you "pre-compute the allDownloadsUri" however if the `info.mId` changes which it can! see method `updateFromDatabase()`